### PR TITLE
Set server time implementation

### PR DESCRIPTION
**Pull request by stefwalter**
Follow on from #1842, finishing up the remaining work. Done:

 * Don't require moment.js, use timedated or date to get server timezone offset
 * Remove useless cockpit.Time function. timedated can tell us server time info

Remaining:

 * Integration tests

### CODE REVIEW

**Comment by stefwalter** [comment]
Added integration test. But it fails, due to a broken implementation.

**Comment by stefwalter** [comment]
@fridex We are not using the timestamp_diff correctly when calling setTime() ... will fix.

**Comment by stefwalter** [comment]
parseInt() is being called incorrectly, it needs the 10 argument, or a leading zero causes it to break.

**Comment by dperpeet** [comment]
I get this error once when switching to NTP time, and once when I switch back to manual

screenshot from 2015-03-19 15 54 00

From manual to manual (just change date/time) I don't get the error.

**Comment by stefwalter** [comment]
@dperpeet Refactored. No longer able to make the above problem resurface. But I'm using Fedora 22, and this may be a race in timedated you're encountering in Fedora 21.

**Comment by stefwalter** [comment]
The 'Access Denied' or 'Operation not permitted' messages are due to SELinux:

https://bugzilla.redhat.com/show_bug.cgi?id=1014315

**Comment by stefwalter** [comment]
@andreasn I fixed the styles of the datepicker, even when used with the old patternfly.

**Comment by stefwalter** [comment]
Finally got through this, fixed implementation. Took forever.

@fridex At the end of the day turns out you have to interpret the typed time on the server itself to get it right. This is because the timezone information and offsetms are different between browser, now, future, past ... and even if one loads moment-timezone.js there's no guarantee that matches what's on the server.

So we do a roundtrip to get the UTC time via 'date' before calling timedate.SetTime()

File: test-avocado/lib/host_setup.sh [hunk 1]
```diff
@@ -28,7 +28,7 @@ HS_BASE_PCKGS="virt-deploy pystache sshpass telnet fabric python-pip avocado vir
 export HS_GRP="virtualization"
 HS_CON="-c qemu:///system"
 export HS_POOLNAME="cockpitx"
-export HS_POOLNAME_PATH=/home/$HS_POOLNAME
+export HS_POOLNAME_PATH=/opt/$HS_POOLNAME
```
Comment (dperpeet) [inline test-avocado/lib/host_setup.sh]: Did you intend to commit this? If so, I think it deserves to be in its own commit, since it changes all test setups.

Comment (stefwalter) [inline test-avocado/lib/host_setup.sh]: Nope didn't mean to.

**Comment by stefwalter** [comment]
Removed accidental host_setup.sh change.

**Comment by dperpeet** [comment]
In Firefox: If I open the datepicker, resize the browser window (while datepicker is open) and then hit Escape, the dialog is closed but the datepicker remains open until I click somewhere inside the window.
